.circleci/config.yml

```diff
@@ -96,6 +96,20 @@ jobs:
             - ./source
             - ./conv.xsl
 
+  # Check circular dependencies with cpp-dependencies tool
+  check_circular_deps:
+    <<: *defaults
+    steps:
+      - attach_workspace:
+          at: /hpx
+      - run:
+          name: Check circular dependencies with cpp-dependencies
+          command: |
+              ./cpp-dependencies --dir /hpx/source/libs /hpx/tools/cpp-dependencies/circular_deps.dot
+              if [[ $(wc -l /hpx/tools/cpp-dependencies/circular_deps.dot | awk '{print $1}') -gt 2 ]]; then exit 1; fi
+      - store_artifacts:
+          path: /hpx/tools/cpp-dependencies/circular_deps.dot
```

Comment:
Should we store a rendered dot file instead?

Reply:
I kept the dot file to check if there is some graph cycles by file length and added a svg generation to visualize the graph :)